File: src/main/java/nl/waterjeloen/quoridor/GUI.java
package nl.waterjeloen.quoridor;

import javax.swing.*;
import java.awt.Dimension;

public class GUI extends JFrame {
    public static final long serialVersionUID = -1L;
    
    private final BoardPanel panel;

    public GUI(Board board) {
        this.panel = new BoardPanel(board);

        setContentPane(panel);
        setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);
        setPreferredSize(new Dimension(1000, 1000));
        pack();
    }

    public BoardPanel getPanel() {
        return panel;
    }

    public void showMessage(String message) {
        JOptionPane.showMessageDialog(this, message);
    }
}

File: src/main/scala/nl/zwaaltjes/quoridor/GUI.scala
package nl.zwaaltjes.quoridor

import java.awt.Dimension
import javax.swing.{JFrame, JOptionPane, UIManager, WindowConstants}

class GUI(board: Board) extends JFrame {
  val panel: BoardPanel = new BoardPanel(board)

  UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName)

  setContentPane(panel)
  setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE)
  setPreferredSize(new Dimension(1000, 1000))
  pack()

  def showMessage(message: String): Unit =
    JOptionPane.showMessageDialog(this, message)
}